### PR TITLE
[Android] Include Margin when measuring from LayoutManager

### LINKED ISSUE #2761: [Android] Invalid Ad Control layout
## Current behavior

A Google Ad control does not layout properly.

## How to reproduce it (as minimally and precisely as possible)

TBD
https://dev.azure.com/nventive/Umbrella/_workitems/edit/176391

## Environment

Nuget Package: 

Package Version(s):

Affected platform(s):

- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by davidjohnoliver**
The LayoutManager is used when `AreDimensionsConstrained = true` for a particular view, which can be set manually as an optimization if it's known that a view's size will never change. It wasn't taking the Margin into account when measuring the view, which was causing errors after layouting changes.

GitHub Issue (If applicable): fixes #2761

## PR Type

- Bugfix

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_FrameworkElement.cs [hunk 1]
```diff
@@ -353,4 +377,16 @@ protected override Size MeasureOverride(Size availableSize)
...
 		}
 	}
+
+	public partial class AspectRatioView : FrameworkElement
+	{
+		public double AspectRatio { get; set; } = 1.5;
+
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			var height = double.IsPositiveInfinity(availableSize.Height) ? 0 : availableSize.Height;
```
[the remark this answers is not in the document]
Comment (davidjohnoliver) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_FrameworkElement.cs]: wut

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.1

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.1

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.1

**Pull request merged**